src/Core/Retry/RetryStrategyInterface.php

```diff
@@ -24,12 +24,18 @@
  */
 interface RetryStrategyInterface
 {
+    /**
+     * Returns the maximum number of retries allowed.
+     *
+     * @return int
+     */
+    public function getMaxRetries(): int;
```

Comment:
Normally adding to interfaces is not backwards compatible. I doubt that this interface has been implemented by userland code though.

Reply:
`Cake\Database\Retry\ReconnectStrategy` is marked internal so I thought the whole mechanism was internal.

Reply:
Looking at the Orcale driver - the only other code I can think of to customize this - doesn't seem to have any strategy code.